src/js/contentscripts/socialwidgets.js

```diff
@@ -238,20 +245,70 @@ function reinitializeWidgetAndUnblockTracker(name) {
   });
 }
 
+/**
+ * Similar to reinitializeWidgetAndUnblockTracker() above,
+ * but also reruns scripts defined in scriptSelectors.
+ *
+ * @param {String} name the name/type of this widget (Disqus, Google reCAPTCHA)
+ */
+function replaceWidgetAndReloadScripts(name) {
+  unblockTracker(name, function () {
+    // restore all widgets of this type
+    WIDGET_ELS[name].forEach(data => {
+      data.parent.replaceChild(data.widget, data.replacement);
+      reloadScripts(data.scriptSelectors, data.fallbackScriptUrl);
+    });
+    WIDGET_ELS[name] = [];
+  });
+}
+
+/**
+ * Find and replace script elements with their copies to trigger re-running.
+ */
+function reloadScripts(selectors, fallback_script_url) {
+  let scripts = document.querySelectorAll(selectors.join(','));
+
+  // if there are no matches, try a known script URL
+  if (!scripts.length && fallback_script_url) {
+    let parent = document.documentElement,
+      replacement = document.createElement("script");
+    replacement.src = fallback_script_url;
+    parent.insertBefore(replacement, parent.firstChild);
+    return;
+  }
+
+  for (let i = 0; i < scripts.length; i++) {
+    let scriptEl = scripts[i];
+
+    // reinsert script elements only
+    if (!scriptEl.nodeName || scriptEl.nodeName.toLowerCase() != 'script') {
+      continue;
+    }
+
+    let replacement = document.createElement("script");
+    for (let j = 0, atts = scriptEl.attributes, n = atts.length; j < n; j++) {
+      replacement.setAttribute(atts[j].nodeName, atts[j].nodeValue);
+    }
+    scriptEl.parentNode.replaceChild(replacement, scriptEl);
+    // reinsert one script and quit
+    break;
+  }
+}
```

Comment:
could we do a `.forEach()` loop here instead? It would be a bit more readable and compact in my opinion.

maybe something like:
` scripts.forEach((script) => { ... }) `
although then the nested for loop within that block might be better off staying the classic ES5 version, since it's tracking a few different variables?

Reply:
You can't short-circuit `forEach` loops, but I could try a `for ... of`

Reply:
Looks like `for .. of` is supported by all the browsers that PB is, could still be worth doing here for readability and removing that extra variable declaration of `let scriptEl = scripts[i]`

It's just a nit though, and there are plenty of examples of old school for loops throughout the rest of the PB code